src/sonic-yang-models/yang-models/sonic-flex_counter.yang

```diff
@@ -0,0 +1,63 @@
+module sonic-flex_counter {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-flex_counter";
+    prefix flex_counter;
+
+    description "FLEX COUNTER YANG Module for SONiC OS";
+
+    revision 2020-04-10 {
+        description "First Revision";
+    }
+
+    container sonic-flex_counter {
+
+        container FLEX_COUNTER_TABLE {
+
+            /* typedef specific to FLEX_COUNTER_TABLE */
+            typedef flex_status {
+                type enumeration {
+                    enum enable;
+                    enum disable;
+                }
+            }
+
+            description "FLEX_COUNTER_TABLE part of config_db.json";
+
+            container PFCWD {
+                leaf FLEX_COUNTER_STATUS {
+                    type flex_status;
+                }
+            }
+
+            container PORT {
+                leaf FLEX_COUNTER_STATUS {
+                    type flex_status;
+                }
+            }
+
+            container QUEUE {
+                leaf FLEX_COUNTER_STATUS {
+                    type flex_status;
+                }
+            }
+
+            container PG_WATERMARK {
+                leaf FLEX_COUNTER_STATUS {
+                    type flex_status;
+                }
+            }
+
+            container QUEUE_WATERMARK {
+                leaf FLEX_COUNTER_STATUS {
+                    type flex_status;
+                }
+            }
+
+        }
```

Comment:
```
orchagent/flexcounterorch.cpp:    {"PORT", PORT_STAT_COUNTER_FLEX_COUNTER_GROUP},
orchagent/flexcounterorch.cpp:    {"PORT_RATES", PORT_RATE_COUNTER_FLEX_COUNTER_GROUP},
orchagent/flexcounterorch.cpp:    {"PORT_BUFFER_DROP", PORT_STAT_COUNTER_FLEX_COUNTER_GROUP},
orchagent/flexcounterorch.cpp:    {"QUEUE", QUEUE_STAT_COUNTER_FLEX_COUNTER_GROUP},
orchagent/flexcounterorch.cpp:    {"PFCWD", PFC_WD_FLEX_COUNTER_GROUP},
orchagent/flexcounterorch.cpp:    {"QUEUE_WATERMARK", QUEUE_WATERMARK_STAT_COUNTER_FLEX_COUNTER_GROUP},
orchagent/flexcounterorch.cpp:    {"PG_WATERMARK", PG_WATERMARK_STAT_COUNTER_FLEX_COUNTER_GROUP},
orchagent/flexcounterorch.cpp:    {BUFFER_POOL_WATERMARK_KEY, BUFFER_POOL_WATERMARK_STAT_COUNTER_FLEX_COUNTER_GROUP},
orchagent/flexcounterorch.cpp:    {"RIF", RIF_STAT_COUNTER_FLEX_COUNTER_GROUP},
orchagent/flexcounterorch.cpp:    {"RIF_RATES", RIF_RATE_COUNTER_FLEX_COUNTER_GROUP},
orchagent/flexcounterorch.cpp:    {"DEBUG_COUNTER", DEBUG_COUNTER_FLEX_COUNTER_GROUP}
```

Reply:
@Iguohan:  one clarification, This is BUFFER_POOL_WATERMARK or BUFFER_POOL_WATERMARK_KEY?